honeybee_3dm/config.py

```diff
@@ -0,0 +1,432 @@
+"""Validation and schema generation for the config file and helper functions."""
+
+import enum
+import json
+import os
+import rhino3dm
+
+from pydantic import BaseModel, validator, Field
+from typing import Dict
+
+from .material import mat_to_dict
+from honeybee.face import Face
+from honeybee.shade import Shade
+from honeybee.aperture import Aperture
+from honeybee.door import Door
+from honeybee.facetype import face_types
+from honeybee.typing import clean_and_id_string, clean_string
+
+
+class FaceObjects(str, enum.Enum):
+    """List of acceptable string names for Honeybee Classes."""
+    door = 'door'
+    shade = 'shade'
+    aperture = 'aperture'
+
+
+class FaceType(str, enum.Enum):
+    """List of acceptable string names for Honeybee face types."""
+    wall = 'wall'
+    roof = 'roof'
+    floor = 'floor'
+    airwall = 'airwall'
+
+
+class GridSettings(BaseModel):
+    """Config for the grid-settings."""
+
+    grid_size: float = Field(
+        1.0,
+        description='Grid spacing.'
+    )
+
+    grid_offset: float = Field(
+        0.0,
+        description='Grid offset'
+    )
+
+    @validator('grid_size')
+    def check_grid_size(cls, v):
+        grid_size = v
+        if grid_size <= 0.0:
+            raise ValueError('Grid size must be greater than zero.')
+        return v
+
+
+class LayerConfig(BaseModel):
+    """Config for layer keys."""
+
+    exclude_from_rad: bool = Field(
+        False,
+        description='Boolean to indicate if this layer should be excluded from '
+        'radiance model.'
+    )
+
+    include_child_layers: bool = Field(
+        False,
```

Comment:
I have made this key optional itself. 

I would request you to re-think this. Imagine a layer structure like the following image;
![layer structure](https://user-images.githubusercontent.com/20086924/101827884-d97ec480-3afe-11eb-8b50-c191d9d108e6.PNG)
If we keep to true by default then a lot of child layers will be unnecessary will be imported. If the user does not want it, they will have to change the setting and will have to re-run the script. I imagine the config file to work like a chisel rather than a hammer. Working on specific parts of the file and not touching the rest.

Reply:
Hmmm... I'm not sure why you think it's unnecessary to import them. As a matter of fact they `must` be imported in this specific case. I still think we should import all the child layers by default.

Reply:
Complied